setup.py

```diff
@@ -28,12 +28,8 @@
     readme = readme_file.read()
 
 dependencies = [
-    "google-api-core >= 1.22.0",
-    "proto-plus >= 0.4.0",
-    "google-cloud-pubsub >= 2.1.0",
-    "grpcio",
-    "setuptools",
-    "overrides",
+    "google-cloud-pubsub >= 2.1.0, <3.0.0dev",
+    "overrides>=2.0.0, < 3.0.0dev",
```

Comment:
I removed `google-api-core`, `proto-plus`, `grpcio`, `setuptools` as those are already installed via `google-cloud-pubsub`.

Having the dependencies declared in both places with version ranges can result in conflicting dependencies. Asn as an example,`proto-plus>=0.4.0` here and [`proto-plus>=1.7.1` in `google-cloud-pubsub`](https://github.com/googleapis/python-pubsub/blob/469ebaa3c449c881089dfc657da5902c1d031803/setup.py#L34) is technically conflicting and results in newer versions of pip raising an error. https://pip.pypa.io/en/stable/user_guide/#fixing-conflicting-dependencies

@googleapis/api-pubsub Please review!

Reply:
Thanks @busunkim96 ! Your changes LGTM.